src/measure_theory/integration.lean

```diff
@@ -519,6 +590,96 @@ end
 
 end measure
 
+section finsupp
+
+variables [measure_space α] [has_zero β] [has_zero γ]
+
+open finset ennreal
+
+/-- A function that vanishes outside a set of finite measure.
+    For simple functions, having finite support is usually the same as being integrable. -/
+def finsupp (f : α → β) : Prop := ∀b ≠ 0, volume (f ⁻¹' {b}) < ⊤
```

Comment:
I find this definition a bit confusing given `data.finsupp`. Over there `finsupp` is defined as a bundled function. Does it make sense to try to make their API's behave the same way? Of is `data.finsupp` too much of a special case of this version?

Reply:
I don't understand this definition. For instance, the identity of real numbers has finite support with this definition. In the way it is written, I think it only makes sense for simple functions, so you should only define it there. If you want to define it for a general function, it should really be `volume (f ⁻¹' (univ - {0})) < ⊤`.
And finite is not the right word, finite volume support is better. `finvolsupp`? Or `fin_vol_supp`?

Given how you use it below, I would keep your definition but only for simple functions. And define it as `simple_function.fin_vol_supp`, and use it as `h : f.fin_vol_supp`.